frontend/pages/sign-in/code.tsx

```diff
@@ -0,0 +1,217 @@
+
+import { useSignIn } from 'services/authentication/authService';
+import { useAcceptInvitation, useInvitedUser } from 'services/invitation/invitationService';
+
+export const getServerSideProps = withLocalizedRequests(async (context) => {
+  // Property 'query' does not exist on type 'GetStaticPropsContext<ParsedUrlQuery, PreviewData>'
+  /** @ts-ignore */
+  const { locale, query } = context;
+  //Redirect back to sign in if the required queries are missing
+  if (!query.password || !query.email) {
+    return { redirect: { destination: Paths.SignIn, permanent: false } };
+  }
+
+  return {
+    props: {
+      intlMessages: await loadI18nMessages({ locale }),
+    },
+  };
+});
+
+type ProjectDeveloperProps = {};
+
+const CODE_LENGTH = 6;
+
+const SignInCode: PageComponent<ProjectDeveloperProps, NakedLayoutProps> = () => {
+  const { formatMessage } = useIntl();
+  const { replace, query } = useRouter();
+  const signIn = useSignIn();
+  const { user, isLoading: isUserLoading } = useMe();
+  const acceptInvitation = useAcceptInvitation();
+  const { invitedUser } = useInvitedUser(query.invitation_token as string);
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    setFocus,
+    setValue,
+  } = useForm<SignInCodeForm>({
+    shouldUseNativeValidation: true,
+    reValidateMode: 'onChange',
+    mode: 'onChange',
+  });
+
+  // Redirect if user is signed-in
+  useEffect(() => {
+    if (!isUserLoading && !!user) {
+      if (user?.role === UserRoles.Light) {
+        // If the user don't have an account go to the 'choose account type' page
+        replace(Paths.AccountType);
+      } else {
+        // If the user has an account go to the dashboard
+        replace(Paths.Dashboard);
+      }
+    }
+  }, [isUserLoading, query.email, query.password, replace, user]);
+
+  // Create an array of numbers from 0 to the code's length
+  const code = Array.from({ length: CODE_LENGTH }, (_, i) => i);
+
+  useEffect(() => {
+    // Set focus at the first input box
+    setFocus('otp_attempt.0');
+  }, [setFocus]);
+
+  const handleChangeCode = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.currentTarget;
+    const currentInputIndex = Number(name.split('.')[1]);
+    let nextInput = currentInputIndex + 1;
+    if (!value || currentInputIndex === CODE_LENGTH - 1) return;
+    setFocus(`otp_attempt.${nextInput}`);
+  };
+
+  const handleSignIn = useCallback(
+    (data: SignIn) => {
+      signIn.mutate(data, {
+        onSuccess: () => {
+          if (!!invitedUser) {
+            acceptInvitation.mutate(query.invitation_token as string, {
+              onSuccess: () => replace(Paths.Dashboard),
+            });
+          } else {
+            replace(Paths.Dashboard);
+          }
+        },
+      });
+    },
+    [signIn, invitedUser, acceptInvitation, query.invitation_token, replace]
+  );
+
+  const handlePasteCode = (ev: ClipboardEvent<HTMLInputElement>) => {
```

Comment:
I think we're missing one instruction in this callback. There's a case where the user is not able to proceed after pasting the code:

1. Write a letter in the first input

A validation error is displayed.

2. Delete that letter
3. Paste the code in the first input

The user is not able to submit the form. They have to manually type the first digit again.